src/components/VirtualizedDataTable/VirtualizedDataTable.tsx

```diff
@@ -0,0 +1,222 @@
+        <Loader />
+      </div>
+    )
+  }
+
+  if (isEmptyTable) {
+    return (
+      <div className={`${CSS_PREFIX}__state-container`} style={{ height }}>
+        <EmptyState />
+      </div>
+    )
+  }
+
+  const virtualItems = rowVirtualizer.getVirtualItems()
+  const totalSize = rowVirtualizer.getTotalSize()
+
+  return (
+    <div
+      className={`${CSS_PREFIX}__container`}
+      ref={containerRef}
+      style={{ height }}
+      role='table'
+      aria-label={ariaLabel}
+    >
+      <Table className='Layer__UI__Table__ProfitAndLossDetailReport'>
+
+        <TableHeader
+          columns={columns.map(col => ({
+            id: col.id,
+            header: col.header,
+            isRowHeader: col.isRowHeader ?? false,
+          }))}
+        >
+          {({ id, header, isRowHeader }) => (
+            <TableColumn
+              key={id}
+              isRowHeader={isRowHeader}
+              className={classNames(
+                `${CSS_PREFIX}__header-cell`,
+                `Layer__UI__Table-Column__${componentName}--${id}`,
+              )}
+            >
+              {header}
+            </TableColumn>
+          )}
+        </TableHeader>
+        <TableBody>
+          <Row
+            className={`${CSS_PREFIX}__spacer`}
+            style={{ height: totalSize }}
+          />
+          {virtualItems.map((virtualRow) => {
+            const rowData = processedData[virtualRow.index]
+            return (
+              <VirtualizedDataTableRow
+                key={rowData.id}
+                rowData={rowData}
+                virtualRow={virtualRow}
+                rowVirtualizer={rowVirtualizer}
+                componentName={componentName}
+                cssPrefix={CSS_PREFIX}
+                columns={columns}
+              />
+            )
+          })}
+        </TableBody>
+      </Table>
+    </div>
+  )
+}
+
+interface VirtualizedDataTableRowProps<TData extends { id: string }, TColumns extends string> {
+  rowData: TData
+  virtualRow: VirtualItem
+  rowVirtualizer: Virtualizer<HTMLDivElement, HTMLTableRowElement>
+  componentName: string
+  cssPrefix: string
+  columns: Column<TData, TColumns>[]
+}
+
+const VirtualizedDataTableRow = <TData extends { id: string }, TColumns extends string>({
+  rowData,
+  virtualRow,
+  rowVirtualizer,
+  componentName,
+  cssPrefix,
+  columns,
+}: VirtualizedDataTableRowProps<TData, TColumns>) => (
+  <Row
+    className={classNames(
+      'Layer__table-row',
+      `${cssPrefix}__row`,
+    )}
+    data-index={virtualRow.index}
+    ref={node => node && rowVirtualizer.measureElement(node)}
+    style={{
+      transform: `translateY(${virtualRow.start}px)`,
+    }}
+  >
+    {columns.map(column => (
+      <Cell
+        key={column.id}
+        className={classNames(
+          'Layer__table-cell',
+          `${cssPrefix}__cell`,
+          `Layer__UI__Table-Cell__${componentName}--${column.id}`,
```

Comment:
I think we can ditch the `Layer__table` prefixed styles everywhere.
```suggestion
        className={classNames(
          `${cssPrefix}__cell`,
          `Layer__UI__Table-Cell__${componentName}--${column.id}`,
```